aten/src/ATen/native/Memory.cpp

```diff
@@ -19,21 +24,40 @@ int64_t _debug_has_internal_overlap(const Tensor& self) {
   return static_cast<int64_t>(at::has_internal_overlap(self));
 }
 
-// Technically, we could force backends to explicitly say "no, we don't support
-// pinned memory, always return false", but this makes life a little easier when
-// you haven't loaded the backend extension at all (which can happen, e.g., on a
-// CPU build of PyTorch and you try to check if something is CUDA pinned)
-bool is_pinned_default(const Tensor& self, c10::optional<Device> device) {
-  return false;
+bool is_pinned(const Tensor& self) {
+  // Only CPU tensors can be pinned
+  if (!self.is_cpu()) {
+    return false;
+  }
+  // Use getAcceleratorHooksInterface to make is_pinned device-agnostic
+  return at::globalContext().isPinnedPtr(self.storage().data());
 }
 
-Tensor pin_memory(const Tensor& self, c10::optional<Device> device) {
-  // Kind of mad that I have to do two dynamic dispatches here, pretty
-  // annoying
-  if (self.is_pinned(device)) {
+Tensor pin_memory(const Tensor& self) {
+  if (self.is_pinned()) {
     return self;
   }
-  return at::_pin_memory(self, device);
+  TORCH_CHECK(self.device().is_cpu(), "cannot pin '", self.toString(), "' only dense CPU tensors can be pinned");
```

Comment:
I think it's better to use _pin_memory here.

Reply:
Since I noticed that the different backends' implement of `_pin_memory` is mainly at host side and  the code's logic is common, where the main difference is in how to get their own Allocator. So I remove function `_pin_memory` and don't go a redispatching way.

Reply:
Hey!
I'm afraid you're going to have to keep `_pin_memory` and the reason is from what is happening in native_functions.yaml.

Reply:
ok, have added it back.